tests/covscripts/coverage.py

```diff
@@ -0,0 +1,118 @@
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2023 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+
+import json
+import os
+import argparse
+
+
+def check_export(filename):
+  """check if the file was successfully exported"""
+  status = os.path.exists(filename)
+  return status
+
+
+def curcov(file_path_curcov_json):
+  """extract current coverage from a json file"""
+  with open(file_path_curcov_json) as curcov_json:
+    curcov_data = json.load(curcov_json)
+  curcov_percentage = curcov_data['totals']['percent_covered']
+  curcov_percentage_ftdt = round(curcov_percentage, 2)
+  return curcov_percentage_ftdt
+
+
+def develop(root_dir, coverage_file_name_txt, coverage_file_name_json):
+  """run and compare coverage for develop branch"""
```

Comment:
Is this function just creating the .txt file and not doing any comparing?

Reply:
And if it is comparing branches, which branches are being compared here? develop to what?

Reply:
Gotcha, it'd help if the function name/docstring makes this clear. Also, the if-else checks to see if coverage_file_name_txt got created successfully -- are they necessary? I don't see why the file would not get created.